performance-tests/bench_2/worker/main.cpp

```diff
@@ -268,6 +269,79 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[]) {
 
     do_wait(config.start_time, "start");
 
+    Log::log() << "Starting Discovery Check." << std::endl;
+
+    process_start_discovery_time = Builder::get_time();
+
+    if (config.wait_for_discovery) {
+      if (config.wait_for_discovery_seconds > 0) {
+
+        std::chrono::seconds timeoutPeriod(config.wait_for_discovery_seconds);
+        std::chrono::system_clock::time_point timeout_time;
```

Comment:
I think we want to be calculating one global absolute timeout up front (currently it recalculates per reader / writer) and then we can pass that timeout in to each reader / writer listener. So this can be const and can do the calculation below (timeout_time = now + timeoutPeriod).

Reply:
So are you looking to pass the timeoutPeriod to the wait_for_expected_match and then perform the now + timeoutPeriod in wait_for_expected_match?

Reply:
No, I want to calculate it once up front (line 278?) and use the same (const?) timeout time for every call. No need to recalculate it for every entity.